pkgs/development/libraries/nss/default.nix

```diff
@@ -126,6 +133,11 @@ in stdenv.mkDerivation rec {
     chmod 0755 $out/bin/nss-config
   '';
 
+  postInstall = stdenv.lib.optionalString useP11kit ''
+    # Replace built-in trust with p11-kit connection
+    ln -sf ${p11-kit}/lib/pkcs11/p11-kit-trust.so $out/lib/libnssckbi.so
```

Comment:
So, how do we gurantee that the ABI between this "fake" output and the one defined in the header files matches? Is that defined in the PKCS#11 standard?

Reply:
It's a good question: the library is [designed](https://p11-glue.github.io/p11-glue/p11-kit/manual/trust-nss.html) to be used like this, so I assume it's ABI compatible with nss, but I'm not sure.

Even if it is, this could be an issue if nss introduces an ABI incompatible change and Nixpkgs is updated before p11-kit. It should be noted p11-kit is maintained and used by red hat, so it should be update quickly enough.